pages/page.py

```diff
@@ -5,6 +5,34 @@
 
 class Page(object):
 
-    def __init__(self, testsetup):
-        self.testsetup = testsetup
-        self.selenium = testsetup.selenium
+    _url = None
+
+    def __init__(self, base_url, selenium):
+        self.base_url = base_url
+        self.selenium = selenium
+
+    def open(self):
+        self.selenium.get(self.url)
+        self.wait_for_page_to_load()
+        return self
+
+    @property
+    def url(self):
+        if self._url is not None:
+            return self._url.format(base_url=self.base_url)
+        return self.base_url
+
+    def wait_for_page_to_load(self):
+        return self
```

Comment:
Just wondering why you are returning `self` here, rather than just using `pass`. It doesn't look like `wait_for_page_to_load` is expecting a return value (not judging by https://github.com/mozilla/mozwebqa-examples/pull/4/files#diff-96842c1cc01108395761c2cd5092f511R16 anyway).